modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.name-edit-field"),
+        ).toBeInTheDocument();
+      });
+
+      const inputElement = screen.getByTestId("templates.name-edit-field");
+      expect(inputElement).toHaveValue("New Template");
+      expect(
+        screen.getByText(new Date().toLocaleDateString()),
+      ).toBeInTheDocument();
+    });
+
+    it("Checks template can be created", async () => {
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("templates.new-template");
+      act(() => templateEdit.click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.name-edit-field"),
+        ).toBeInTheDocument();
+      });
+
+      const inputElement = screen.getByTestId("templates.name-edit-field");
+      await userEvent.clear(inputElement);
+      await userEvent.type(inputElement, "Test Template");
+
+      expect(inputElement).toHaveValue("Test Template");
+
+      const saveButton = screen.getByTestId("templates.save-template");
+      expect(saveButton).toBeInTheDocument();
+
+      act(() => saveButton.click());
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("templates.name-edit-field"),
+        ).not.toBeInTheDocument();
+      });
+
+      expect(screen.getByText("Test Template"));
+    });
+
+    it("Checks input field when edit button is pressed", async () => {
+      render(<Templates />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("templates.edit");
+      act(() => templateEdit[0].click());
+
+      expect(
+        screen.getByTestId("templates.name-edit-field"),
+      ).toBeInTheDocument();
+    });
+
+    it("Checks template can be edited", async () => {
+      render(<Templates />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("templates.edit");
+      act(() => templateEdit[0].click());
+
+      const inputElement = screen.getByTestId("templates.name-edit-field");
+      await userEvent.clear(inputElement);
+      await userEvent.type(inputElement, "New Template Name");
+
+      expect(inputElement).toHaveValue("New Template Name");
+
+      const saveButton = screen.getByTestId("templates.save-template");
+      expect(saveButton).toBeInTheDocument();
+
+      act(() => saveButton.click());
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("templates.name-edit-field"),
+        ).not.toBeInTheDocument();
+      });
+
+      expect(screen.getByText("New Template Name"));
```

Comment:
Would add toast success message appears after calling the API (Noted, toast is mocked, is easy to check, look at other test cases)